pkg/api/convertertoagentpoolonlyapi.go

```diff
@@ -353,10 +353,16 @@ func convertV20180331AgentPoolOnlyProperties(obj *v20180331.Properties) *Propert
 	if obj.ServicePrincipalProfile != nil {
 		properties.ServicePrincipalProfile = convertV20180331AgentPoolOnlyServicePrincipalProfile(obj.ServicePrincipalProfile)
 	}
+
 	if obj.AddonProfiles != nil {
 		properties.AddonProfiles = convertV20180331AgentPoolOnlyAddonProfiles(obj.AddonProfiles)
 	}
 
+	// TODO: validate that API is v20180331
```

Comment:
I left his comment here so we wouldn't forget, but IDK where such validation should actually go.

Reply:
`convertV20180331AgentPoolOnlyProperties` is invoked for API version 2018-03-31 only so I removed the comment. I don't think we need separate validation for this.